homeassistant/components/dsmr/sensor.py

```diff
@@ -409,3 +444,105 @@ def unit_of_measurement(self):
         unit = self.get_dsmr_object_attr("unit")
         if unit:
             return f"{unit}/{TIME_HOURS}"
+
+
+class Timer:
+    """async Timer object with reset.
+
+    a resettable timer is not present in asyncio, however this integration
+    needs one to timeout faulty connections. It includes
+    garbage collection uses loop.call_later.
+
+    """
+
+    def __init__(self, delay, timer_callback, *args, loop=None):
+        """Create a timer with confingurable timeout and callback.
+
+        Has no timerhandle  on creation(will be created on start) and the
+        callback and delay will be used to create this timerhandle.
+
+        """
+        self.timerhandle = None
+        self.set_timeout(delay)
+        self.set_callback(timer_callback)
+        self.args = args
+        if loop:
+            self.loop = loop
+        else:
+            self.loop = asyncio.get_event_loop()
+
+    def set_timeout(self, delay):
+        """Set the timeout of the timer to a certain delay in float seconds.
+
+        will not change the timeout of a running timer.
+
+        """
+        self.delay = delay
+
+    def set_callback(self, timer_callback):
+        """Set the callback method for the timer to call when it runs out.
+
+        Will not change the callback of a running timer.
+
+        """
+        if callable(timer_callback):
+            self.timer_callback = timer_callback
+        else:
+            raise Exception("callback for Timer is not callable")
+
+    def start(self):
+        """Start the timer by creating a task to be called on."""
+        if not self._timer_active():
```

Comment:
Use guard clause and don't raise `Exception` but instead raise more appropriate errors like `ValueError` or `RuntimeError`.

```python
if self._timer_active():
    raise RuntimeError("Timer already started")
```